rootex/framework/components/physics/physics_component.cpp

```diff
@@ -0,0 +1,201 @@
+#include "physics_component.h"
+#include "core/resource_loader.h"
+#include "framework/systems/physics_system.h"
+
+#include "entity.h"
+
+PhysicsComponent::PhysicsComponent(const String& matName, float volume)
+    : m_MaterialName(matName)
+    , m_MotionState(Matrix::Identity)
+	, m_Material(0, 0)
+    , m_Volume(volume)
+{
+	m_TransformComponent = nullptr;
+	physicsMaterial = ResourceLoader::CreateLuaTextResourceFile("game/assets/config/physics.lua");
+	physicsLua.loadExecuteScript(physicsMaterial);
```

Comment:
This should be done only once in the entire game. Move this to GamePhysics constructor